src/boot/efi/boot.c

```diff
@@ -2337,6 +2357,46 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
 
         config_load_defaults(&config, root_dir);
 
+        if (setup_mode_enabled() && config.auto_enroll) {
+                Print(L"Secure Boot auto enrollement is enabled.\n");
+
+                /* Read the required files in order to perform auto-enrollement */
+                err_sb_db  = file_read(root_dir, L"\\db.auth", 0, 0, &sb_db_buffer, &sb_db_size);
+                err_sb_kek = file_read(root_dir, L"\\KEK.auth", 0, 0, &sb_kek_buffer, &sb_kek_size);
+                err_sb_pk  = file_read(root_dir, L"\\PK.auth", 0, 0, &sb_pk_buffer, &sb_pk_size);
+
+                if (!EFI_ERROR(err_sb_db) && !EFI_ERROR(err_sb_kek) && !EFI_ERROR(err_sb_pk)) {
```

Comment:
i think we should enroll what we can, and continue otherwise.

Reply:
Along with your comment about setting the default to automatic enrollment, I think it might be fairly 'dangerous' in some corner cases.

If for some reason the user is in setup mode and for whatever reason only the PK is available on the ESP then the users' system will become unbootable (adding PK switches on secure boot enforcement but no db is available) until put in setup mode again.

It is probably not that big of a deal with a workstation however it might be a problem on servers, you might have someone turn on the setup mode physically and install the system at a later time only to lock up your system and need another manual intervention.

I'm not sure if that is a big concern but that was my reasoning for making auto_enroll default to false and only allow 'atomic' (db+kek+pk) enrollments.